conjure-java-core/src/main/java/com/palantir/conjure/java/services/ServiceGenerators.java

```diff
@@ -17,45 +17,63 @@
 package com.palantir.conjure.java.services;
 
 import com.google.common.base.Strings;
+import com.palantir.conjure.java.util.ErrorGenerationUtils;
 import com.palantir.conjure.java.util.Javadoc;
+import com.palantir.conjure.java.util.Packages;
 import com.palantir.conjure.spec.EndpointDefinition;
+import com.palantir.javapoet.ClassName;
+import com.palantir.javapoet.CodeBlock;
+import com.palantir.javapoet.MethodSpec;
 import java.util.Optional;
 import java.util.stream.Collectors;
 
 public final class ServiceGenerators {
+    public record EndpointJavaDocGenerationOptions(boolean includeRequestLine, boolean includeEndpointErrors) {}
```

Comment:
We could use an enum instead of booleans for clarity e.g.

```java
enum RequestLineJavadoc {
    INCLUDE,
    EXCLUDE
}
```

methods/ctors which take multiple booleans are easy to get wrong, and difficult to read. IDE features help, but the github UI isn't so kind

Reply:
Yep that's cleaner. Updated.